### PR TITLE
Show degree subject HESA code in support section

### PR DESCRIPTION
**Pull request by malcolmbaig**
## Context

We want to start displaying HESA codes in the support section to easily check if the feature is working as we roll it out.
## Changes proposed in this pull request
- Add a subject column to the qualifications table and move the subject over from the degree column.
- Add the degree type and subject HESA codes (if available) to the table.

Screenshot_20200629_144656

## Link to Trello card
https://trello.com/c/PYf7wdSf

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Comment by malcolmbaig** [comment]
Raising as a draft PR to run the visual change past @paulrobertlloyd . Code feedback still welcome.

@paulrobertlloyd does that screenshot above look OK as a way for us to start displaying the subject HESA code?

**Comment by paulrobertlloyd** [comment]
Looks good, though I’d be more explicit, changing the column title to `Degree (HESA code)`.

**Comment by paulrobertlloyd** [comment]
Was `Masters (Design)` manually entered? If we were to show `Degree (HESA code)` in the column heading, we’d want to avoid using brackets elsewhere.

**Comment by malcolmbaig** [comment]
@paulrobertlloyd latest set of commits look like the following. The `Masters/Arts` entry is freeform input, the `BA/Product Design`  is HESA selections. The grade HESA codes haven't been implemented yet but will display in a similar way once they're in.

Screenshot_20200629_134518

**Comment by paulrobertlloyd** [comment]
Love it! I’d keep the HESA codes as small grey text as you had initially, but other than that, 👍

**Comment by malcolmbaig** [comment]
> +1 Code looks good and it works for me locally.
> 
> One question about this feature (not this PR): Do we need to update the test application generator to include HESA codes or does this just work?

We'd need to add them in. Currently HESA codes are only populated if degrees are updated via one of the forms (`DegreeTypeForm`, `DegreeSubjectForm`, and, imminently, `DegreeInstitutionForm`/`DegreeGradeForm`).

File: app/components/qualification_subject_component.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class QualificationSubjectComponent < ViewComponent::Base
+  def initialize(qualification:)
+    @qualification = qualification
+  end
+
+  def subject
+    @qualification.subject&.titleize
```
Comment (duncanjbrown) [inline app/components/qualification_subject_component.rb:7]: Do we allow this to be nil? We validate :subject on GcseQualificationTypeForm, OtherQualificationForm and DegreeSubjectForm

File: app/lib/hesa/degree_type.rb [hunk 2]
```diff
@@ -1,6 +1,10 @@
 module Hesa
   class DegreeType
-    DegreeTypeStruct = Struct.new(:hesa_code, :abbreviation, :name, :level)
+    DegreeTypeStruct = Struct.new(:hesa_code, :abbreviation, :name, :level) do
+      def shortest_display_name
```
Comment (duncanjbrown) [inline app/lib/hesa/degree_type.rb:4]: Nice!

File: app/components/qualification_subject_component.rb (hunk 1 above)
Comment (malcolmbaig) [reply app/components/qualification_subject_component.rb:7]: We do, yes. A candidate can abandon the 'add degree' flow partway through, meaning this table may need to render a partially-completed degree.

**Pull request merged**